src/os_maild/os_maild_client.c

```diff
@@ -744,12 +745,16 @@ void PrintTable(cJSON *item, char *printed, size_t body_size, char *tab, int cou
         log_size = strlen(key) + strlen(tab) + strlen(item->string) + strlen(delimitator) + strlen(endline);
 
         if (body_size > log_size) {
-            strncat(printed, tab, strlen(tab));
-            strncat(printed, item->string, strlen(item->string));
-            strncat(printed, delimitator, strlen(delimitator));
+            strncat(printed, tab, body_size);
```

Comment:
Hi @RamiroRD 

In fact, the behavior is the same, the only difference is that I am using the size of the destination buffer and decrementing it when concatenating the new string. The warning solved was because the previous implementation was wrong, the size in `strncat` never has to depend on the length of the source buffer.

Best regards
Hanes

Reply:
This makes sense, I get it now. Thanks